src/main/java/seedu/parser/Parser.java

```diff
@@ -122,6 +122,12 @@ public Command parseCommand(String userInput) {
                 return new CheckCommand(args);
             } catch (IncompleteCommandException e) {
                 return new IncorrectCommand(CheckCommand.COMMAND_WORD + CheckCommand.COMMAND_DESCRIPTION);
+            } catch (NumberFormatException e) {
+                return new IncorrectCommand(Command.INCORRECT_COST_FORMAT);
+            } catch (IllegalArgumentException e) {
+                return new IncorrectCommand(Command.INCORRECT_ENUM_TYPE);
+            } catch (DateTimeParseException e) {
+                return new IncorrectCommand(Command.INVALID_DATE_MESSAGE);
```

Comment:
Actually, I removed these catch statements in CheckCommand because these exceptions shouldn't be thrown to Parser?